### PR TITLE
Update Haml syntax and gem to version 6

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

- Closes #11920 

Haml 6 is 1.7x faster than Haml 5 and actively maintained. Our wide use of the attribute hash syntax for non-standard AngularJS attributes starting with `ng-` (not `data-`) made the upgrade challenging though. Since we had over 800 lines of code to change, I opted for writing a script to rewrite our haml templates automatically. I hope that I will have time to share this code with the wider community, as gem or as rubocop integration.

#### What should we test?

- Everything. :scream: 
- We can't test everything but I may have missed some attributes which can make Javascript fail in random places. During development, I found that the admin area was more affected than the shop.
- I would suggest to go through the registration, checkout, order management, order cycle management and product management.
- And keep the dev tools console open to spot any additional JS errors that weren't there before.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by rioug — APPROVED** [review]
Good work ! I didn't look too closely at the script as it's well covered by the test.

File: app/assets/javascripts/templates/admin/panels/enterprise_package.html.haml
Comment (dacook) [inline app/assets/javascripts/templates/admin/panels/enterprise_package.html.haml:97]: Great to see the nesting was handled beautifully!

File: app/assets/javascripts/templates/admin/schedule_dialog.html.haml
Comment (dacook) [inline app/assets/javascripts/templates/admin/schedule_dialog.html.haml:24]: I see here that all attributes for the element were re-written.
It converts the clean symbol notation to string/rocket notation which isn't necessary.

But.. maybe we can just look away because we'd like to say bye bye Angular anyway..

**Review by dacook — CHANGES_REQUESTED** [review]
Great work!

It's a shame that it took so much work to upgrade the markup files, so I definitely agree you should put this in a gem and become famous!

One little thing: Given that the files are getting bigger, it would be nice to use the shorter symbol-like syntax with a colon, instead of the rocket (`=>`) syntax.  I think it would be relatively simple to change, what do you think?

File: lib/haml_up.rb
Comment (mkllnk) [reply lib/haml_up.rb]: Initially I put this in the too hard basket because I would need to find the difference between simple symbols like `:disabled` and symbols needing quotes like `:"ng-model"` and then re-arranging the colon for the hash syntax. But now I thought about it again and remembered the simple `to_symbol` which can then leverage `inspect` as before. Still need to move the colon but that's easily done. Have a look at my added commits.

**Review by dacook — APPROVED** [review]
Thanks for delving back into this, it makes the code (and the PR diff) look way cleaner now!
Clever update, I didn't know it could be so simple 🏅

**Comment by filipefurtad0** [comment]
Hey @mkllnk ,

I've tested all admin sections, probably not going into _all_ sub-sections, but still, I've tested:
- order management
- order cycle management
- product management
- customer management
- reports
- enterprise management [*]

Placed backoffice orders via new order page; tested subscriptions too, by creating/running a new subscription.

[*] Spotted only a minor glitch was hard to reproduce (seems to have occurred when attempting to edit an enterprise which has granted rights, but which the logged in user is not an owner), it relates to what it seems to be a missing spinner:

Bugnsag [here](https://app.bugsnag.com/yaycode/open-food-network-js-uk/errors/649dfa0721e646000811e5b4?filters[error.status]=open&filters[event.since]=30d&filters[app.release_stage]=staging&event_id=65e87c9200dd96dc7cce0000).
```
 Uncaught (in promise) Error: Missing target element "spinner" for "primary-details" controller
    get stimulus.js:3372
    value primary_details_controller.js:31
    promise callback*value primary_details_controller.js:29
    value primary_details_controller.js:9
    value stimulus.js:554
    value stimulus.js:497
    value stimulus.js:75
    __trace__ inline-script-content.js:149
    r inline-script-content.js:162
    value stimulus.js:46
    value stimulus.js:212
    value stimulus.js:202
    value stimulus.js:193
    value stimulus.js:152
    value stimulus.js:1593
    value stimulus.js:1624
    value stimulus.js:1488
    value stimulus.js:1383
    value stimulus.js:1369
    value stimulus.js:1368
    value stimulus.js:1347
    value stimulus.js:871
    value stimulus.js:795
    value stimulus.js:678
    value stimulus.js:640
    value stimulus.js:833
    value stimulus.js:1317
    value stimulus.js:1458
    value stimulus.js:1551
    value stimulus.js:2252
    value stimulus.js:2482
    value stimulus.js:3051
    value stimulus.js:2919
    value stimulus.js:1488
    value stimulus.js:1383
    value stimulus.js:1369
    value stimulus.js:1368
    value stimulus.js:1347
    value stimulus.js:871
    value stimulus.js:795
    value stimulus.js:678
    value stimulus.js:640
    value stimulus.js:833
    value stimulus.js:1317
    value stimulus.js:1458
    value stimulus.js:2883
    value stimulus.js:2989
    e stimulus.js:3153
    d stimulus.js:1
    M stimulus.js:1
    k stimulus.js:1
    i stimulus.js:2
    s stimulus.js:3
    promise callback*i stimulus.js:2
    s stimulus.js:3
    n stimulus.js:3
    n stimulus.js:3
    value stimulus.js:3162
    value stimulus.js:3248
    <anonymous> index.js:12
    Webpack 5
stimulus.js:3372:14
    get stimulus.js:3372
    value primary_details_controller.js:31
    (Async: promise callback)
    value primary_details_controller.js:29
    value primary_details_controller.js:9
    value stimulus.js:554
    value stimulus.js:497
    value stimulus.js:75
    __trace__ inline-script-content.js:149
    (Async: EventListener.handleEvent)
    r inline-script-content.js:162
    value stimulus.js:46
    value stimulus.js:212
    value stimulus.js:202
    value stimulus.js:193
    value stimulus.js:152
    value stimulus.js:1593
    value stimulus.js:1624
    value stimulus.js:1488
    value stimulus.js:1383
    value stimulus.js:1369
    forEach self-hosted:203
    value stimulus.js:1368
    value stimulus.js:1347
    value stimulus.js:871
    value stimulus.js:795
    value stimulus.js:678
    value stimulus.js:640
    value stimulus.js:833
    value stimulus.js:1317
    value stimulus.js:1458
    value stimulus.js:1551
    value stimulus.js:2252
    value stimulus.js:2482
    value stimulus.js:3051
    value stimulus.js:2919
    value stimulus.js:1488
    value stimulus.js:1383
    value stimulus.js:1369
    forEach self-hosted:203
    value stimulus.js:1368
    value stimulus.js:1347
    value stimulus.js:871
    value stimulus.js:795
    value stimulus.js:678
    value stimulus.js:640
    value stimulus.js:833
    value stimulus.js:1317
    value stimulus.js:1458
    value stimulus.js:2883
    value stimulus.js:2989
    e stimulus.js:3153
    d stimulus.js:1
    M stimulus.js:1
    k stimulus.js:1
    i stimulus.js:2
    s stimulus.js:3
    (Async: promise callback)
    i stimulus.js:2
    s stimulus.js:3
    n stimulus.js:3
    n stimulus.js:3
    value stimulus.js:3162
    value stimulus.js:3248
    <anonymous> index.js:12
    Webpack 5

```

Customer/shopfront side:
- registered
- checked out
- saved Stripe cards (checkout)
- added new cards
- changed shop/order cycle

Very tempted to merge this one as is, but will wait for your feedback on the missing spinner.

**Comment by filipefurtad0** [comment]
Hum, I just noticed, not all subscription emails are arriving. There are some associated bugsnag errors:
https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/65e89906911d290008ffa165?filters[error.status]=open&filters[event.since]=30d&event_id=65e89e2b00dc6332dc070000

This does not seem related to the Haml change (right? Hamls should impact the front end? )... but I'll leave it to your judgement.

**Comment by mkllnk** [comment]
> a missing spinner

@filipefurtad0 Did you try to reproduce this in production? I don't see how this would have changed in this PR.

**Comment by filipefurtad0** [comment]
> Did you try to reproduce this in production? I don't see how this would have changed in this PR.

I've tried other servers with master, without success. I've concluded the data is quite specific in this case, so I've tried again on the same server, but with some other PR staged: it is reproducible - which proves the point that it is not introduced by this PR.

Thanks @mkllnk !

Merging.

**Pull request merged**
